perfkitbenchmarker/benchmarks/fio_benchmark.py

```diff
@@ -20,24 +20,62 @@
 
 import json
 import logging
+import posixpath
+import re
+
+import jinja2
 
 from perfkitbenchmarker import data
 from perfkitbenchmarker import flags
 from perfkitbenchmarker import vm_util
 from perfkitbenchmarker.packages import fio
 
+LOCAL_JOB_FILE_NAME = 'fio.job'  # used with vm_util.PrependTempDir()
+REMOTE_JOB_FILE_PATH = posixpath.join(vm_util.VM_TMP_DIR, 'fio.job')
+DEFAULT_TEMP_FILE_NAME = 'fio-temp-file'
+MAX_FILE_SIZE_GB = 100
+DISK_USABLE_SPACE_FRACTION = 0.9
+
+
+# This dictionary maps scenario names to dictionaries of fio settings.
+SCENARIOS = {
+    'sequential_write': {
+        'name': 'sequential_write',
+        'rwkind': 'write',
+        'blocksize': '512k'
+    },
+    'sequential_read': {
+        'name': 'sequential_read',
+        'rwkind': 'read',
+        'blocksize': '512k'
+    },
+    'random_write': {
+        'name': 'random_write',
+        'rwkind': 'randwrite',
+        'blocksize': '4k'
+    },
+    'random_read': {
+        'name': 'random_read',
+        'rwkind': 'randread',
+        'blocksize': '4k'
+    }
+}
+
 
 FLAGS = flags.FLAGS
 
-flags.DEFINE_string('fio_benchmark_filename', 'fio_benchmark_file',
-                    'scratch file that fio will use')
-flags.DEFINE_string('fio_jobfile', 'fio.job', 'job file that fio will use')
-flags.DEFINE_integer('memory_multiple', 10,
-                     'size of fio scratch file compared to main memory size.')
+flags.DEFINE_string('fio_jobfile', None,
+                    'Job file that fio will use. Cannot use with '
+                    '--generate_scenarios.')
+flags.DEFINE_list('generate_scenarios', None,
+                  'Generate a job file with the given scenarios. If no '
+                  'scenarios are listed, generate all of them. Cannot use '
+                  'with --fio_jobfile.')
 flags.DEFINE_boolean('against_device', False,
-                     'Test direct against scratch disk. If set True, will '
-                     'create a modified job file in temp directory, '
-                     'which ignores directory and filename parameter.')
+                     'If --generate_scenarios is given, test against the raw '
```

Comment:
I think the documentation here is confusing. From the code, whenever against_device is True, the disk is umounted.